tex/solutions.tex

```diff
@@ -0,0 +1,280 @@
+\end{itemize}
+\end{solution}
+
+\begin{solution}[to \cref{exer:iso_in_posetcategory}]
+First, let $(X,\leq_Y)$ be a preorder. A morphism $f:x\to y$ is an isomorphism if and only if there exists a morphism $g: y\to x$ such that $\co{f}{g}=\Id[x]$ and $\co{g}{f}=\Id[y]$. But, in a preorder category, each hom-set has a unique element if it is non-empty. So, for any $g\in \Hom{y}{x}$ and $f\in \Hom{x}{y}$, we always have $\co{f}{g}=\Id[x]$ and $\co{g}{f}=\Id[y]$. Hence a morphism $f:x\to y$ in a preorder-category is an isomorphism if and only if there exists a morphism $g:y\to x$. The existence of a morphism $f:x\to y$ means precisely that $x\leq y$. Hence, isomorphisms in a preorder-category corresponds with a pair of elements $(x,y)$ in $X$ such that $x\leq y$ and $y\leq x$.\\
+If $(X,\leq_X)$ is a poset, i.e. satisfies antisymmetry, then if $x\leq y$ and $y\leq x$, we must have $x=y$. Consequently, in a poset-category, the only isomorphisms are the identity morphisms (i.e. corresponding with $x\leq x$).
+\end{solution}
+
+\begin{solution}[to \cref{ex:mono-inj}]
+The monomorphisms in $\SET$ correspond precisely with the injective functions, i.e. the functions $f:X\to Y$ which satisfy
+\[
+\forall x_1,x_2\in X: f(x_1)=f(x_2) \implies x_1=x_2,
+\]
+Indeed:
+\begin{itemize}
+\item Assume $f$ is injective. Let $g,h: Z\to X$ be functions such that $\co{g}{f}=\co{h}{f}$. We have to show $g=h$, i.e. 
+\[
+\forall z\in Z: g(z)=h(z).
+\]
+Since $f$ is injective, it suffices to show 
+\[
+\forall z\in Z: f(g(z))=f(h(z)).
+\]
+But this holds by the condition of $g$ and $h$. Hence, $f$ is indeed a monomorphism.
+\item Assume $f$ is a monomorphism. We have to show that for each $x_1,x_2\in X$, we have $f(x_1) = f(x_2)$. Let $\mathbf{1} = \{\star\}$ be a singleton set and define
+\[
+g_1 : \mathbf{1}\to X: \star\mapsto x_1,\quad  g_2 : \mathbf{1}\to X: \star\mapsto x_2, 
+\]
+Since $f(x_1)=f(x_2)$, we have $\co{g_1}{f} = \co{g_2}{f}$. But $f$ is a monomorphism, hence $g_1 = g_2$ which means $x_1 = g_1(\star) = g_2(\star) = x_2$. Thus, $f$ is indeed injective.
+\end{itemize}
+\end{solution}
+
+\begin{solution}[to \cref{exer:sections_in_set_injective}]
+Let $f:X\to Y$ be a section with a retraction $h:Y\to X$, i.e. $\co{f}{h} = \Id[Y]$. By \cref{ex:mono-inj}, it suffices to show that $f$ is a monomorphism. Let  $g_1, g_2 : Z \to X$ be morphisms in $\SET$ such that $\co{g_1}{f} = \co{g_2}{f}$. We have to show $g_1=g_2,$, this follows from the following computation:
+\begin{eqnarray*}
+g_1 =& \co{g_1}{\Id[Y]} & \text{ by unit law},\\ 
+	=& \co{g_1}{\co{f}{h}} & \text{ since $f$ section},\\ 
+	=& \co{\co{g_1}{f}}{h} & \text{ by associativity},\\
+	=&  \co{\co{g_2}{f}}{h} & \text{ by assumption},\\
+	=& \co{g_2}{\co{f}{h}} & \text{ by associativity},\\ 
+	=& \co{g_2}{\Id[Y]} & \text{ since $f$ section},\\
+	=& g_2 & \text{ by unit law}.
+\end{eqnarray*}
+Notice that this proof shows that in an arbitrary category, a section is always a monomorphism.
```

Comment:
This remark is maybe better positioned after the exercise?

Reply:
But this remark only makes sense when considering the solution, so after the exercise in the solution file or in the CT4P file after the exercise?

Reply:
After the exercise, you can say "The solution given in \cref{solution:foobar} actually works for any category. Hence we can state the following lemma:"

This is just a suggestion, anything is good with me. But I think it is best if general statements are given in the main text, not in the solutions.